src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/ObjectApiTest.cs

```diff
@@ -0,0 +1,224 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Dfs;
+using Catalyst.Abstractions.Hashing;
+using Catalyst.Core.Lib.Dag;
+using Catalyst.Core.Modules.Dfs.Tests.Utils;
+using Catalyst.Core.Modules.Hashing;
+using MultiFormats.Registry;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Catalyst.Core.Modules.Dfs.Tests.IntegrationTests.CoreApi
+{
+    public class ObjectApiTest
+    {
+        private readonly IHashProvider _hashProvider;
+        private IDfsService ipfs;
+
+        public ObjectApiTest(ITestOutputHelper output)
+        {
+            _hashProvider = new HashProvider(HashingAlgorithm.GetAlgorithmMetadata("sha2-256"));
+            ipfs = TestDfs.GetTestDfs(output, null, null, _hashProvider);
+        }
+
+        [Fact]
+        public async Task New_Template_Null()
+        {
+            var node = await ipfs.ObjectApi.NewAsync();
+            Assert.Equal("QmdfTbBqBPQ7VNxZEYEj14VmRuZBkqFbiwReogJgS1zR1n", node.Id);
+        }
+
+        [Fact]
+        public async Task New_Template_UnixfsDir()
+        {
+            var node = await ipfs.ObjectApi.NewAsync("unixfs-dir");
+            Assert.Equal("QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn", node.Id);
+
+            node = await ipfs.ObjectApi.NewDirectoryAsync();
+            Assert.Equal("QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn", node.Id);
+        }
+
+        [Fact]
+        public void New_Template_Unknown()
+        {
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = ipfs.ObjectApi.NewAsync("unknown-template").Result;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)